sonaranalyzer-dotnet/rspec/cs/S4432_c#.html

```diff
@@ -0,0 +1,36 @@
+<p>Encryption algorithms can be used with various modes. Some combinations are not secured:</p>
+<ul>
+  <li> Electronic Codebook (ECB) mode: Under a given key, any given plaintext block always gets encrypted to the same ciphertext block. Thus, it does
+  not hide data patterns well. In some senses, it doesn't provide serious message confidentiality, and it is not recommended for use in cryptographic
+  protocols at all. </li>
+  <li> Cipher Block Chaining (CBC) with PKCS#5 padding (or PKCS#7) is susceptible to padding oracle attacks. CBC + PKCS#7 can be used if combined with
+  an authenticity check (HMAC-SHA256 for example) on the cipher text. </li>
+</ul>
+<p>In both cases, Galois/Counter Mode (GCM) with no padding should be preferred. As the .NET framework doesn't provide this natively, the use of a
+certified third party lib is recommended. </p>
+<p>This rule raises an issue when any of the following CipherMode is detected: ECB, CBC, OFB, CFB, CTS.</p>
+<h2>Noncompliant Code Example</h2>
+<pre>
+AesManaged aes = new AesManaged
+{
+  KeySize = 128,
+  BlockSize = 128,
+  Mode = CipherMode.OFB, // Noncompliant
+  Padding = PaddingMode.PKCS7
+};
+</pre>
+<h2>Compliant Solution</h2>
+<pre>
+Cipher c = Cipher.getInstance("AES/GCM/NoPadding");
```

Comment:
Java-style naming of _getInstance_. Does this matter?

Reply:
Yes... We don't have any compliant solution for C# xD

Reply:
I asked Alex to move the compliant solution to a sub-task so we don't end up with the inherited solution.